mpas_analysis/shared/climatology/climatology.py

```diff
@@ -346,19 +353,38 @@ def compute_climatologies_with_ncclimo(config, inDirectory, outDirectory,
     if remappedDirectory is not None:
         args.extend(['-O', remappedDirectory])
 
-    print 'running: {}'.format(' '.join(args))
-
-    # make sure any output is flushed before we add output from the
-    # subprocess
-    sys.stdout.flush()
-    sys.stderr.flush()
-
     # set an environment variable to make sure we're not using czender's
     # local version of NCO instead of one we have intentionally loaded
     env = os.environ.copy()
     env['NCO_PATH_OVERRIDE'] = 'No'
 
-    subprocess.check_call(args, env=env)  # }}}
+    if logger is None:
+        print 'running: {}'.format(' '.join(args))
+        # make sure any output is flushed before we add output from the
+        # subprocess
+        sys.stdout.flush()
+        sys.stderr.flush()
+
+        subprocess.check_call(args, env=env)
+    else:
+        logger.info('running: {}'.format(' '.join(args)))
+        for handler in logger.handlers:
+            handler.flush()
+
+        process = subprocess.Popen(args, stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE, env=env)
+        stdout, stderr = process.communicate()
+
+        if stdout:
+            logger.info(stdout)
+        if stderr:
```

Comment:
> One option would be simply to remove the ERROR: at the beginning so there's no difference in the logger whether something is an error, a warning or a regular print (and you would need to explicitly include this in the statement instead).

would that mean that it would have to remove this `if stderr` statement here?

Reply:
No, I would remove the text "ERROR:" from here:
https://github.com/MPAS-Dev/MPAS-Analysis/pull/259/files#diff-f4c01dd9c50e825c99d4d138d9001e07R435
We do still want to capture stderr, we just don't necessarily want to explicitly print ERROR: in front of every line.

By the way, I don't think the example above is where the particular warning (displayed as error) is coming from.  Instead, it's a warning inside numpy somewhere that gets redirected to the logger using the following trick:
https://github.com/MPAS-Dev/MPAS-Analysis/pull/259/files#diff-f4c01dd9c50e825c99d4d138d9001e07R246

Reply:
I see.
Yes, I would definitely remove the 'ERROR:' from there and customize our print statements.